Components/GoogleAnalyticsOptIn/script.js

```diff
@@ -0,0 +1,152 @@
+import $ from 'jquery'
+import Cookies from 'js-cookie'
+
+class GoogleAnalyticsOptIn extends window.HTMLDivElement {
+  constructor (self) {
+    self = super(self)
+    self.$ = $(self)
+    self.props = self.getInitialProps()
+    self.setOptions()
+    self.resolveElements()
+    self.bindFunctions()
+    self.bindEvents()
+    return self
+  }
+
+  setOptions () {
+    this.cookieName = 'acceptedTracking'
+  }
+
+  getInitialProps () {
+    let data = {}
+    try {
+      data = JSON.parse($('script[type="application/json"]', this).text())
+    } catch (e) {}
+    return data
+  }
+
+  resolveElements () {
+    this.$acceptButton = $('[data-accept]', this)
+    this.$declineButton = $('[data-decline]', this)
+    this.$optOutLink = $('.googleOptOutLink')
+    this.gaId = this.isValidId(this.props.gaId) ? this.props.gaId : false
+    this.disableStr = 'ga-disable-' + this.props.gaId
+  }
+
+  bindFunctions () {
+    this.addGTAGFunction = this.addGTAGFunction.bind(this)
+    this.addGTAGFunctionFallback = this.addGTAGFunctionFallback.bind(this)
+    this.addGTAGScript = this.addGTAGScript.bind(this)
+    this.acceptTracking = this.acceptTracking.bind(this)
+    this.declineTracking = this.declineTracking.bind(this)
+    this.showModal = this.showModal.bind(this)
+    this.hideModal = this.hideModal.bind(this)
+  }
+
+  bindEvents () {
+    this.$acceptButton.on('click', this.acceptTracking)
+    this.$declineButton.on('click', this.declineTracking)
+    this.$optOutLink.on('click', this.showModal)
```

Comment:
how about adding the click event listener like that?
$(body).on('click', '.googleOptOutLink',this.showModal)
That way it would also work when the link is somehow dynamically added to the dom at any point in time

Reply:
I've added this and removed the $optoutlink propery.